shared/middleware/channels_api.rb

```diff
@@ -164,7 +164,7 @@ class ChannelsApi < Sinatra::Base
   # Marks the specified channel as published.
   #
   post %r{/v3/channels/([^/]+)/publish/([^/]+)} do |channel_id, project_type|
-    not_authorized if under_13?
+    not_authorized if under_13? && !%w(artist playlab).include?(project_type)
```

Comment:
I don't know much about our 'publish' action. Is it sufficient to trust the passed `project_type` here?  What happens if a student under 13 publishes a channel with an applab project but passes `artist` as the type?

There's even a spot in the test you modify below where we check:
```ruby
     project_type = 'foo'
     post "/v3/channels/#{channel_id}/publish/#{project_type}", {}.to_json, 'CONTENT_TYPE' => 'application/json;charset=utf-8'
     assert last_response.ok?
```
...which seems wrong to me.  I'd expect either a 404 since `/v3/channels/{channel}/publish/foo` doesn't seem like a valid URL, or a 400 since there's an invalid argument.  So putting aside that we allow users over 13 to publish with any string as the project type (security?) is there any more reliable way for us to verify that an under-13 user is allowed to publish the project in question?